icu4c/source/i18n/complexunitsconverter.cpp

```diff
@@ -0,0 +1,86 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+#include <utility>
+
+#include "cmemory.h"
+#include "complexunitsconverter.h"
+#include "uassert.h"
+#include "unicode/fmtable.h"
+#include "unitconverter.h"
+
+U_NAMESPACE_BEGIN
+
+ComplexUnitsConverter::ComplexUnitsConverter(const MeasureUnit inputUnit, const MeasureUnit outputUnits,
+                                             UErrorCode &status) {
+    auto singleUnits = outputUnits.splitToSingleUnits(status);
+    MaybeStackVector<MeasureUnit> singleUnitsInOrder;
```

Comment:
I think you only want to do this splitting if `outputUnits` is a sequence unit.  If it is a compound unit, then you don't want to split.

Reply:
done!